File: rand.ts
class Rand {

    static rand_seed: DataView = new DataView(new ArrayBuffer(4));
    static M: number = 0x7FFFFFFF;
    static Seed(seed: number) {
        
        Rand.rand_seed.setUint32(0, seed);
        console.log(Rand.M);
    }

    static Rand():number{ 
        Rand.rand_seed.setUint32(0, (Rand.rand_seed.getUint32(0) * 0x41A7));   
        Rand.rand_seed.setUint32(0, Rand.rand_seed.getUint32(0) % Rand.M);
        return Rand.rand_seed.getUint32(0);
    }
    
    static RandMax(max:number): number {
        return Rand.Rand() % max;
    }
    
}

File: rand.go
package rand

import (
	"time"
)

var rand_seed uint32

func init() {
	rand_seed = uint32(time.Now().Unix())
}

func SetSeed(seed uint32) {
	rand_seed = seed
}

func Rand() uint32 {
	rand_seed = (rand_seed * uint32(0x41A7)) % ((1 << 31) - 1)
	return rand_seed
}

//随机0 ~ max-1 之间的数
func RandMax(max uint32) uint32 {
	return Rand() % max
}
